### DISCUSSION TITLE
Only list todos in certain context/project

### DISCUSSION THREAD

**Discussion by fritzrehde** [todotxt/todo.txt-cli#366] [Q&A] [answered]
I am looking for a way to print only the todos from a certain context/project. As far as I understand it, `listcon` and `listproj` only list the name of all contexts/projects but not the todos themselves. Does the feature I am looking for exist?

**Comment by inkarkat** [comment, marked as answer]
Yes, `listcon` and `listproj` only get you the used contexts / projects. Filtering of items is done via `ls` / `list`, by passing the optional `TERM` argument.

Usually, the project and context names are simple and rather unique alphanumeric words, and you can simple pass them directly, e.g. `todo.sh ls +you`, but to be totally correct, you'd have to assert the whitespace before the sigil: `todo.sh ls \ +you` or `todo.sh ls [[:space:]]+you`.

This will match any context / project starting with the `TERM`; i.e. `+you` will also match inside `+youtube`. In order to just match the full context / project, we need another assertion for either whitespace or the end of the line at the end: `todo.sh ls '[[:space:]]+you\([[:space:]]\|$\)'`

For interactive queries, I'm usually lazy and just specify the name as-is; any additional undesired tasks can be mentally skipped. The ugly correct filtering is important for scripting uses, though, that's why I've mentioned them.
